src/rabbit_extended_api.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Add missing functionality to rabbitmq_admin.AdminAPI."""
+
+import json
+import urllib
+
+import rabbitmq_admin
+
+
+class ExtendedAdminApi(rabbitmq_admin.AdminAPI):
```

Comment:
I think this is fine to get what we need from the Python binding but please can we look to submit this to the rabbitmq_admin project as well.

Reply:
Although I just realised how long it is since there was a release update.

Reply:
Yeah, that's what I found, it doesn't seem to be actively maintained anymore. If I was writing the charm from scratch I don't think I would use this library. It is quite lightweight so should be easy to switch out in future.